crates/server/src/store/recursor/authority.rs

```diff
@@ -205,12 +205,15 @@ impl LookupObject for RecursiveLookup {
         None
     }
 
-    fn dnssec_validated(&self) -> bool {
-        // TODO research what spec / other impls do when the answer section is empty, DNSSEC
-        // validation is enabled but nameservers provided no NSEC3 records
-        self.0
-            .records()
-            .iter()
-            .all(|record| record.proof().is_secure())
+    fn dnssec_summary(&self) -> DnssecSummary {
+        let proofs = self.0.records().iter().map(|record| record.proof());
```

Comment:
Would suggest formulating as follows:

```rust
let mut secure = true;
for record in self.0.records.iter() {
    match record.proof() {
        Proof::Secure => (),
        Proof::Bogus => return DnsssecSummary::Bogus,
        Proof::Insecure => secure = false,
    }
}

match secure {
    true => DnssecSummary::Secure,
    false => DnssecSummary::Insecure,
}
```

This avoids iterating through records multiple times and IMO makes the logic more clear.